src/Components/Instagram-authentication.js

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import { auth } from "../firebase";
+
+export default class InstagramAuth extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      emailInputValue: "",
+      passwordInputValue: "",
+      newUser: true,
+      errorCode: "",
+      errorMessage: "",
+    };
```

Comment:
Let's define a defaultState object with these values, which you can then use here and in line 29 as well.
```suggestion
    this.state = defaultState
```